ingestion/src/metadata/ingestion/source/messaging/kinesis/metadata.py

```diff
@@ -146,6 +146,20 @@ def _get_topic_partitions(self, topic_name: str) -> List[str]:
             )
         return all_partitions
 
+    def yield_topic_sample_data(
+        self, topic_details: BrokerTopicDetails
+    ) -> TopicSampleData:
+        """
+        Method to Get Sample Data of Messaging Entity
+        """
+        if self.context.topic and self.generate_sample_data:
+            yield OMetaTopicSampleData(
+                topic=self.context.topic,
+                sample_data=self._get_sample_data(
+                    topic_details.topic_name, topic_details.topic_metadata["partitions"]
```

Comment:
not sure if we should try catch the full call here

Reply:
I'll add the try catch

Reply:
it is added